rxbindings/src/test/java/com/amplifyframework/rx/RxAdaptersTest.java

```diff
@@ -0,0 +1,185 @@
+            .toCompletable((onResult, onError) -> onResult.accept(RandomString.string()))
+            .test()
+            .assertComplete()
+            .assertNoErrors();
+    }
+
+    /**
+     * The {@link Completable} returned by {@link RxAdapters#toCompletable(VoidResultEmitter)}
+     * will dispatch an error when {@link VoidResultEmitter}'s error consumer is invoked.
+     */
+    @Test
+    public void completableFiresErrorWhenErrorEmitted() {
+        Throwable expected = new Throwable(RandomString.string());
+        RxAdapters
+            .toCompletable((onResult, onError) -> onError.accept(expected))
+            .test()
+            .assertError(expected)
+            .assertNotComplete();
+    }
+
+    /**
+     * The {@link Completable} returned by {@link RxAdapters#toCompletable(VoidResultEmitter)}
+     * is cancelable.
+     */
+    @SuppressWarnings("checkstyle:WhitespaceAround")
+    @Test
+    public void completableIsCancellable() {
+        Completable completable = RxAdapters.toCompletable(((onResult, onError) -> {}));
+        TestObserver<?> observer = completable.test();
+        observer.cancel();
+        assertTrue(observer.isCancelled());
+    }
+
+    /**
+     * The {@link Single} returned by {@link RxAdapters#toSingle(CancelableResultEmitter)}
+     * will dispatch an error when the {@link CancelableResultEmitter}'s error consumer is
+     * invoked.
+     */
+    @Test
+    public void singleFiresErrorWhenErrorEmitted() {
+        Throwable expected = new Throwable(RandomString.string());
+        RxAdapters
+            .toSingle((onResult, onError) -> {
+                onError.accept(expected);
+                return new NoOpCancelable();
+            })
+            .test()
+            .assertError(expected)
+            .assertNoValues();
+    }
+
+    /**
+     * The {@link Single} returned by {@link RxAdapters#toSingle(CancelableResultEmitter)}
+     * will dispatch a result when the {@link CancelableResultEmitter}'s value consumer
+     * is invoked.
+     */
+    @Test
+    public void singleFiresResultWhenEmitted() {
+        String result = RandomString.string();
+        RxAdapters
+            .toSingle((onResult, onError) -> {
+                onResult.accept(result);
+                return new NoOpCancelable();
+            })
+            .test()
+            .assertValue(result)
+            .assertComplete();
+    }
+
+    /**
+     * {@link RxAdapters#toSingle(CancelableResultEmitter)} returns a {@link Single} which
+     * can be canceled. The Amplify {@link Cancelable} returned by that emitter will also show
+     * as canceled.
+     */
+    @Test
+    public void singleIsCancelable() {
+        SimpleCancelable cancelable = new SimpleCancelable();
+        TestObserver<?> observer = RxAdapters.toSingle(((onResult, onError) -> cancelable))
+            .test();
+        observer.cancel();
+        assertTrue(observer.isCancelled());
+        assertTrue(cancelable.isCanceled());
```

Comment:
Ugh, we should just use `Cancellable` and `Cancelled` everywhere since that's what RX went with when naming [`Cancellable` interface](http://reactivex.io/RxJava/javadoc/io/reactivex/functions/Cancellable.html).

Reply:
Yea, I agree. American English is the only dialect of English that prefers the spelling `Cancelable`. But, I think I researched this one day and determined that it was a non-American who added `Cancellable` into Java. So, I guess they win by virtue of getting their first!

Reply:
For another PR...